### PR TITLE
feat: complete a full request based on example project

### PR DESCRIPTION
**Pull request by DMwangnima**
#### Check the PR title.

- [x] This PR title match the format: \<type\>(optional scope): \<description\>
- [x] The description of this PR title is user-oriented and clear enough for others to understand.
- [x] Attach the PR updating the user documentation if the current PR requires user awareness at the usage level. [User docs repo](https://github.com/cloudwego/cloudwego.github.io)

#### (Optional) More detailed description for this PR(en: English/zh: Chinese).

en:
zh(optional):

### CODE REVIEW

File: pkg/codec.go [hunk 1]
```diff
@@ -126,20 +132,54 @@ func (m *Hessian2Codec) messageServiceInfo(ctx context.Context, message remote.M
...
+	case dubbo.RESPONSE_WITH_EXCEPTION:
+		exception, err := decoder.Decode()
+		if err != nil {
+			return err
+		}
+		if exceptionErr, ok := exception.(error); ok {
+			return exceptionErr
+		}
+		return fmt.Errorf("dubbo side exception: %v", exception)
+	}
```
Comment (felix021) [inline pkg/codec.go:186]: `default: return fmt.Errorf("Unsupported payloadType %v", payloadType)`

File: tests/dubbo/go-client/cmd/client.go [hunk 2]
```diff
@@ -28,7 +28,7 @@ import (
 
 func main() {
 	// 启动框架
-	if err := config.Load(); err != nil {
+	if err := config.Load(config.WithPath("/Users/wangyuxuan/my_project/codec-hessian2/tests/dubbo/go-client/conf/dubbogo.yaml")); err != nil {
```
Comment (felix021) [inline tests/dubbo/go-client/cmd/client.go]: Please use a relative path, and include this yml file in the tests directory

File: tests/dubbo/go-server/cmd/server.go [hunk 3]
```diff
@@ -59,7 +59,7 @@ func init() {
 }
 
 func main() {
-	if err := config.Load(); err != nil {
+	if err := config.Load(config.WithPath("/Users/wangyuxuan/my_project/codec-hessian2/tests/dubbo/go-server/conf/dubbogo.yaml")); err != nil {
```
Comment (felix021) [inline tests/dubbo/go-server/cmd/server.go]: Please use a relative path, and include this yml file in the tests directory

File: tests/kitex/kitex_gen/echo/testservice/client.go [hunk 4]
```diff
@@ -19,7 +19,7 @@ type Client interface {
 func NewClient(destService string, opts ...client.Option) (Client, error) {
 	var options []client.Option
 	options = append(options, client.WithDestService(destService))
-	options = append(options, client.WithPayloadCodec(hessian2.NewHessian2Codec()))
+	options = append(options, client.WithCodec(hessian2.NewHessian2Codec()))
```
Comment (felix021) [inline tests/kitex/kitex_gen/echo/testservice/client.go:22]: Please update `tests/kitex/readme.md` accordingly.

**Review by jasondeng1997 — APPROVED** [review]

**Review by felix021 — APPROVED** [review]

**Review by felix021 — APPROVED** [review]

**Pull request merged**